api/spec/mutations/sign_in_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+module Mutations
+   module Auth
+    RSpec.describe SignIn, type: :request do
+      describe '.resolve' do
+        context 'When the login credentials are correct' do
+          user = FactoryBot.create(:user, email: Faker::Internet.unique.email, )
+        it 'creates a user session' do
+          expect do
+          post '/graphql', params: { query: query(id: user.id) }
+          expect(response).to have_http_status(200)
+          json = JSON.parse(response.body)
+          expect(json['data']['signIn']['token']).not_to be_nil
+         end
```

Comment:
No need of wrapping the data in a large `expect` block.
You can omit line `10` and `15` safely.
You can also have all the `expect` statements occur last in the block and the operations in line 11 and 13 arranged together. Eg.
```ruby
it 'creates a user session' do
  post '/graphql', params: { query: query(id: user.id) }
  json = JSON.parse(response.body)
  expect(response).to have_http_status(200)
  expect(json['data']['signIn']['token']).not_to be_nil
end
```